### PR TITLE
@task PDA-1178 Create decrease stock request message and put it on request queue

### PR DESCRIPTION
**Pull request by tren-parasoft**
Create decrease stock request message and put it on request queue.

### CODE REVIEW

File: src/main/java/com/parasoft/demoapp/dto/InventoryOperation.java [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package com.parasoft.demoapp.dto;
+
+public enum InventoryOperation {
+    INCREASE_FOR_ORDER_DENIAL, DECREASE_FOR_ORDER_CREATION, INCREASE_FOR_ORDER_CREATION_FAILED
```
Comment (yshi-parasoft) [inline src/main/java/com/parasoft/demoapp/dto/InventoryOperation.java:4]: @rwang-parasoft Besides different error message, is there any other reason why we need to differentiate "INCREASE_FOR_ORDER_DENIAL" and "INCREASE_FOR_ORDER_CREATION_FAILED"?

**Comment by yshi-parasoft** [comment]
@rwang-parasoft I'm thinking the enum type for operation should only include "INCREASE" and "DECREASE" because there's no difference to perform a "INCREASE_FOR_ORDER_DENIAL" or "INCREASE_FOR_ORDER_CREATION_FAILED" on inventory service side. We just use it for indicating what type of error message we want to specify later. 
So we are actually mixing the concepts of "operation" and "error".
How about let's just keep "INCREASE" and "DECREASE" for the operation type, but create another enum type just for error code. So it can include a number as error code and replace the "info" argument in "InventoryOperationRequestMessageDTO".
In that way we only map the code with corresponding message on order service side when we need the message without passing the error message along and mixing the concepts of operation and error.

File: src/main/java/com/parasoft/demoapp/model/industry/OrderStatus.java [hunk 2]
```diff
@@ -2,16 +2,26 @@
...
+    SUBMITTED("Submitted", 0),
+    PROCESSED("Processed", 1),
+    CANCELED("Canceled", 1),
+    APPROVED("Approved", 2),
+    DECLINED("Declined", 2);
 
     private String status;
 
-    OrderStatus(String status) {
+    private int code;
```
Comment (yshi-parasoft) [inline src/main/java/com/parasoft/demoapp/model/industry/OrderStatus.java:13]: Let's change "code" to "priority"

**Comment by rwang-parasoft** [comment]
> @rwang-parasoft I'm thinking the enum type for operation should only include "INCREASE" and "DECREASE" because there's no difference to perform a "INCREASE_FOR_ORDER_DENIAL" or "INCREASE_FOR_ORDER_CREATION_FAILED" on inventory service side. We just use it for indicating what type of error message we want to specify later. So we are actually mixing the concepts of "operation" and "error". How about let's just keep "INCREASE" and "DECREASE" for the operation type, but create another enum type just for error code. So it can include a number as error code and replace the "info" argument in "InventoryOperationRequestMessageDTO". In that way we only map the code with corresponding message on order service side when we need the message without passing the error message along and mixing the concepts of operation and error.

I agree with operation and error code, but I think we should keep "info" argument, since it is human readable if there is an exception happends. And the info message is used for the response when use SOAtest to sen a bad request message.

**Comment by rwang-parasoft** [comment]
> @rwang-parasoft I'm thinking the enum type for operation should only include "INCREASE" and "DECREASE" because there's no difference to perform a "INCREASE_FOR_ORDER_DENIAL" or "INCREASE_FOR_ORDER_CREATION_FAILED" on inventory service side. We just use it for indicating what type of error message we want to specify later. So we are actually mixing the concepts of "operation" and "error". How about let's just keep "INCREASE" and "DECREASE" for the operation type, but create another enum type just for error code. So it can include a number as error code and replace the "info" argument in "InventoryOperationRequestMessageDTO". In that way we only map the code with corresponding message on order service side when we need the message without passing the error message along and mixing the concepts of operation and error.

I agree with operation and error code, but I think we should keep "info" argument, since it is human readable if there is an exception happends. And the info message is used for the response when use SOAtest to sen a bad request message.

**Comment by rwang-parasoft** [comment]
> > @rwang-parasoft I'm thinking the enum type for operation should only include "INCREASE" and "DECREASE" because there's no difference to perform a "INCREASE_FOR_ORDER_DENIAL" or "INCREASE_FOR_ORDER_CREATION_FAILED" on inventory service side. We just use it for indicating what type of error message we want to specify later. So we are actually mixing the concepts of "operation" and "error". How about let's just keep "INCREASE" and "DECREASE" for the operation type, but create another enum type just for error code. So it can include a number as error code and replace the "info" argument in "InventoryOperationRequestMessageDTO". In that way we only map the code with corresponding message on order service side when we need the message without passing the error message along and mixing the concepts of operation and error.
> 
> I agree with operation and error code, but I think we should keep "info" argument, since it is human readable if there is an exception happends. And the info message is used for the response when use SOAtest to sen a bad request message.

The final decision after disscuss with Yuqian.

The construction of InventoryOperationRequestMessageDTO is:
{
    operation: INCREASE/DECREASE;
    orderNumber;
    inventoryInfos;
    info: only for exceptions;
}

**Pull request merged**